src/withStyles.jsx

```diff
@@ -19,14 +19,16 @@ export function withStyles(
     stylesPropName = 'styles',
     themePropName = 'theme',
     flushBefore = false,
+    pureComponent = false,
   } = {},
 ) {
   const styleDef = styleFn && ThemedStyleSheet.create(styleFn);
+  const BaseClass = pureComponent ? React.PureComponent : React.Component;
```

Comment:
this should probably either throw if `PureComponent` is not a function, or silently fall back to `Component`